Formula/b/build2.rb

```diff
@@ -0,0 +1,143 @@
+  depends_on "pkgconf"
+
+  uses_from_macos "curl"
+  uses_from_macos "openssl"
+  uses_from_macos "sqlite"
+
+  # These two patches add support for importing .tbd libraries from
+  # .../MacOS*.sdk/usr/lib/ directories (for example, libsqlite3.tbd). Drop
+  # them once upgraded to 0.17.0 which includes them.
+  #
+  patch do
+    url "https://github.com/build2/build2/commit/f221ca94a304abaa79239cf2cbaf3ffafbbc092c.patch?full_index=1"
+    sha256 "391fbd38e863bc3c23f1cddf360f097d210b31ad1e9a906eb583d927f0d216a3"
+    directory "build2"
+  end
+
+  patch do
+    url "https://github.com/build2/build2/commit/17c1214943959b304ab5cdcccacec8a70c7a6c0f.patch?full_index=1"
+    sha256 "a860ce05f75bf653a135c443fab8299e7c0c7e0980cb0f86583a37ba72ecce7c"
+    directory "build2"
+  end
+
+  def install
+    pkgconf = Formula["pkgconf"]
+    sqlite = Formula["sqlite"] if OS.linux?
+
+    # Note that the below steps are equivalent to invoking build.sh with the
+    # --local --private --system libpkgconf,libsqlite3 options. The only
+    # difference are a few customizations to find Homebrew's libpkgconf and
+    # libsqlite3.
+    #
+    # For background on these steps, see:
+    #
+    # https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml
+
+    # Suppress loading of default options files.
+    #
+    ENV["BUILD2_DEF_OPT"] = "0"
+
+    # Bootstrap, stage 1.
+    #
+    # Note that we disable all warnings since we cannot do anything more
+    # granular during bootstrap stage 1.
+    #
+    cd "build2" do
+      system "make",
+             "-j", ENV.make_jobs,
+             "-f", "./bootstrap.gmake",
+             "CXX=#{ENV.cxx}", "CXXFLAGS=-w"
+    end
+
+    # Bootstrap, stage 2.
+    #
+    # Force using Homebrew's libpkgconf rather than building libpkg-config
+    # from source.
+    #
+    # Note that the build2 driver bootstrapped at stage 1 above doesn't read
+    # .pc files. Thus, during the bootstrap stage 2, it is assumed that the
+    # headers and libraries of the system dependencies can be found by the
+    # compiler and linker automatically. Failed that we have to manually
+    # specify their header/library paths, which is what we do here for
+    # libpkgconf.
+    #
+    cd "build2" do
+      system "build2/b-boot", "-v",
+             "-j", ENV.make_jobs,
+             "config.cxx=#{ENV.cxx}",
+             "config.bin.lib=static",
+             "config.build2.libpkgconf=true",
+             "config.cc.poptions=-I#{pkgconf.include}/pkgconf",
+             "config.cc.loptions=-L#{pkgconf.lib}",
+             "build2/exe{b}"
+      mv "build2/b", "build2/b-boot"
+    end
+
+    # Local build.
+    #
+    # Note that we don't build hermetically to avoid baking Homebrew-specific
+    # paths into the final dylibs.
+    #
+    loptions = "-L#{pkgconf.lib}"
+    loptions += " -L#{sqlite.lib}" if OS.linux?
+
+    system "build2/build2/b-boot", "-V",
+           "config.cxx=#{ENV.cxx}",
+           "config.cc.coptions=-O3",
```

Comment:
I don't think Homebrew respects this, it'll be stripped out.

Reply:
You mean `-O3`? I checked the build logs and the option appears to be passed to `clang`. But I can change it to something else if there is something more appropriate.

Reply:
Ah, I see, thanks. I would need to still keep them since they will be included into configurations that will be used by the user to build build-time dependencies. I've added the following note to the formula:

```
    # Note that while Homebrew's clang wrapper will strip any optimization
    # options, we still want to pass them since they will also be included
    # into the ~host and ~build2 configurations that will be used to build
    # built-time dependencies and build system modules, respectively, when
    # the user uses actual clang.
```